### PR TITLE
Optimized various things

### PR DESCRIPTION
**Pull request by clover0916**

### CODE REVIEW

**Review by lb-migii — COMMENTED** [review]
Please answer or rebuttal my question

File: config.yml
Comment (clover0916) [reply config.yml:10]: I'll change that sentence

File: lib/config/config.go
Comment (clover0916) [reply lib/config/config.go:58]: Sry, this code is not good code. I'll change it soon then I push it

File: lib/db/db.go
Comment (clover0916) [reply lib/db/db.go:55]: I am following the best practices of Go lang.

File: lib/db/db.go
Comment (clover0916) [reply lib/db/db.go:195]: I don't think so. This code is correct.

File: lib/handler/message_create.go
Comment (clover0916) [reply lib/handler/message_create.go:41]: I just don't want urlCheck() to run when the command is used, so there is no problem.

Comment (clover0916) [reply lib/handler/message_create.go:41]: I don't see the need to be able to use commands when mentioned.

File: lib/handler/url_check.go
Comment (clover0916) [reply lib/handler/url_check.go:42]: OK. I'll change it

File: lib/handler/url_check.go
Comment (clover0916) [reply lib/handler/url_check.go:46]: OK. I'll change it

File: lib/handler/url_check.go
Comment (clover0916) [reply lib/handler/url_check.go:61]: I think this code is more readable, so I prefer this one.

**Comment by ozraru** [comment]
@clover0916 Don't mark as resolved until Maintainer approve it.

File: lib/handler/url_check.go
Comment (ozraru) [inline lib/handler/url_check.go:57]: What is this?

File: lib/handler/url_check.go
Comment (clover0916) [reply lib/handler/url_check.go:33]: OK

Comment (clover0916) [reply lib/handler/url_check.go:33]: OK

File: lib/handler/message_create.go
Comment (ozraru) [reply lib/handler/message_create.go:41]: > I don't see the need to be able to use commands when mentioned.
It is copy from Vanilla.
It was the way to access command when you forget the prefix.
It can be removed if prefix is not going to be configurable with command.

File: lib/handler/url_check.go
Comment (ozraru) [reply lib/handler/url_check.go:57]: Sorry, my way to comment is bad.
> strings.Contains(result, ".mobile")
> results[i] = strings.ReplaceAll(results[i], ".mobile", "")
I want example that this code to be used.

File: lib/handler/message_create.go
Comment (clover0916) [reply lib/handler/message_create.go:41]: Oh! Sry, I'll fix them.

File: lib/handler/url_check.go
Comment (clover0916) [reply lib/handler/url_check.go:33]: OK

File: lib/handler/url_check.go
Comment (clover0916) [reply lib/handler/url_check.go:57]: I'm sorry. The correct one is m., not mobile.
ex) m.youtube.com/watch?v=xxxx

File: lib/handler/url_check.go
Comment (ozraru) [inline lib/handler/url_check.go:60]: Why do you explicitly check Contains before ReplaceAll?
If it don't contains, ReplaceAll will do nothing.

Comment (clover0916) [reply lib/handler/url_check.go:60]: OK. I understand

File: lib/handler/url_check.go
Comment (ozraru) [inline lib/handler/url_check.go:43]: You means `continue` of outer loop?

File: lib/handler/url_check.go
Comment (ozraru) [inline lib/handler/url_check.go:51]: > results[i] = strings.ReplaceAll(result, "www.", "")

This may be overrided on

> results[i] = result[:len(result)-1]

File: lib/handler/url_check.go
Comment (ozraru) [inline lib/handler/url_check.go:39]: In this case, I think using "standard `for`" is better than using `range`.
Using both of `results[i]` and `result` is confused.

File: main.go
Comment (ozraru) [inline main.go:21]: > `.Error()`

This is not necessary.

File: lib/handler/message_reaction_add.go
Comment (ozraru) [inline lib/handler/message_reaction_add.go:12]: According to discord document, `reaction.Emoji.Name` can be null.
https://discord.com/developers/docs/resources/emoji#emoji-object

**Review by ozraru — CHANGES_REQUESTED** [review]

File: lib/handler/message_reaction_add.go
Comment (clover0916) [reply lib/handler/message_reaction_add.go:12]: I don't know how to deal with this, so for now, I have made a code that returns return in case of null.

File: lib/handler/url_check.go
Comment (clover0916) [reply lib/handler/url_check.go:51]: Sry, I'll fix it

Comment (clover0916) [reply lib/handler/url_check.go:51]: OK

File: lib/handler/url_check.go
Comment (clover0916) [reply lib/handler/url_check.go:39]: I changed some code, please take a look.

File: lib/handler/url_check.go
Comment (clover0916) [reply lib/handler/url_check.go:43]: That code was a mistake and I fixed it.

File: lib/handler/url_check.go
Comment (clover0916) [reply lib/handler/url_check.go:21]: Sorry, my knowledge was insufficient.

File: lib/handler/message_reaction_add.go
Comment (ozraru) [reply lib/handler/message_reaction_add.go:16]: You can check both of `id` and `name`.
So, `reaction.Emoji.ID == config.CurrentConfig.Duplicate.Delete || reaction.Emoji.Name == config.CurrentConfig.Duplicate.Delete` is better.

Comment (clover0916) [reply lib/handler/message_reaction_add.go:16]: Thanks

**Review by lb-migii — CHANGES_REQUESTED** [review]

File: lib/handler/message_reaction_add.go
Comment (ozraru) [inline lib/handler/message_reaction_add.go:10]: Search in state cache first.
If not found, receive using session.

**Review by ozraru — CHANGES_REQUESTED** [review]

File: lib/handler/message_reaction_add.go
Comment (clover0916) [reply lib/handler/message_reaction_add.go:10]: OK

File: lib/handler/message_reaction_add.go
Comment (clover0916) [reply lib/handler/message_reaction_add.go:12]: Fixed

**Pull request closed without merging**
